torch/distributions/constraints.py

```diff
@@ -458,26 +461,37 @@ def check(self, value):
         return _LowerCholesky().check(value) & unit_row_norm
 
 
-class _Symmetric(Constraint):
+class _Square(Constraint):
     """
-    Constrain to Symmetric square matrices.
+    Constrain to square matrices.
     """
     event_dim = 2
 
     def check(self, value):
+        return torch.zeros(value.shape[:-2], dtype=torch.bool).fill_(value.shape[-2] == value.shape[-1])
+
+
+class _Symmetric(_Square):
+    """
+    Constrain to Symmetric square matrices.
+    """
+
+    def check(self, value):
+        assert super().check(value).all(), \
+            "Cannot check symmetricity if the input is not a square matrix."
         return (value.transpose(-2, -1) == value).all(dim=-1).all(dim=-1)
 
 
-class _PositiveDefinite(Constraint):
+class _PositiveDefinite(_Symmetric):
     """
     Constrain to positive-definite matrices.
     """
-    event_dim = 2
 
     def check(self, value):
-        # Assumes that the matrix or batch of matrices in value are symmetric
+        assert super().check(value).all(), \
+            "Cannot check positive definiteness if the input is not a symmetric matrix."
         # info == 0 means no error, that is, it's SPD
-        return torch.linalg.cholesky_ex(value).info.eq(0).unsqueeze(0)
+        return (value.eq(value.mT).all(-2).all(-1) and torch.linalg.cholesky_ex(value).info.eq(0)).unsqueeze(0)
```

Comment:
Same as above. This should now be `super().check(value) and ...` and the assert should be removed.

Reply:
I am not sure about this part! Can you read the comment above and give me advice on final options?

Reply:
How about we break the checks like?

```
sym_check = super().check(value)
if not sym_check.all():
    return sym_check
return (value.eq(value.mT).all(-2).all(-1) and torch.linalg.cholesky_ex(value).info.eq(0))
```
Same for above. The disadvantage here is that the exact place where the validation fails might not be obvious to the user, but the check API remains consistent. Also, as discussed in the other comment, let us remove the `unsqueeze(0)` here.

Reply:
Great to go with this style. I will push another soon.